design/0001-frr.md

```diff
@@ -0,0 +1,411 @@
+# MLB-0001: BGP FRR enablement
+
+## Summary
+
+The purpose of this enhancement is to use [Free Range
+Routing](https://frrouting.org/) (FRR) as an alternative BGP implementation
+in MetalLB. When directed to, MetalLB will publish prefixes via FRR rather than
+MetalLB’s current built-in BGP speaker.
+
+## Motivation
+
+The motivation for this enhancement is to provide an alternative
+production-ready BGP agent for use in MetalLB. Overall, this should reduce the
+maintenance cost for the MetalLB project by reducing the effort required for
+adding additional BGP features and bug fixes. For example, there are a number of
```

Comment:
I'd say the BGP package has been the most stable area of the speaker since I joined the project. Therefore, I'm not sure we can argue that switching to FRR would reduce the project's maintenance burden.

To counter the above, I'd say we (the maintainers) have been slow in adding e.g. new BGP features to MetalLB, which IMO gives merit to the idea of getting such features "for free" by adopting FRR.

Reply:
Ok, I will clarify that in the document.